src/measure_theory/function/strongly_measurable.lean

```diff
@@ -308,24 +434,89 @@ lemma fin_strongly_measurable_iff_strongly_measurable_and_exists_set_sigma_finit
     hf.2.some_spec.2.2⟩
 
+lemma ae_fin_strongly_measurable_zero {α β} {m : measurable_space α} (μ : measure α) [has_zero β]
+  [topological_space β] :
+  ae_fin_strongly_measurable (0 : α → β) μ :=
+⟨0, fin_strongly_measurable_zero, eventually_eq.rfl⟩
+
 namespace ae_fin_strongly_measurable
 
 variables {α β : Type*} {m : measurable_space α} {μ : measure α} [topological_space β]
   {f g : α → β}
 
+section mk
+variables [has_zero β]
+
+/-- A `fin_strongly_measurable` function such that `f =ᵐ[μ] hf.mk f`. See lemmas
+`fin_strongly_measurable_mk` and `ae_eq_mk`. -/
+protected noncomputable def mk (f : α → β) (hf : ae_fin_strongly_measurable f μ) : α → β := hf.some
+
+lemma fin_strongly_measurable_mk (hf : ae_fin_strongly_measurable f μ) :
+  fin_strongly_measurable (hf.mk f) μ :=
+hf.some_spec.1
+
+lemma ae_eq_mk (hf : ae_fin_strongly_measurable f μ) : f =ᵐ[μ] hf.mk f :=
+hf.some_spec.2
+
+protected lemma ae_measurable {β} [has_zero β] [measurable_space β] [metric_space β] [borel_space β]
+  {f : α → β} (hf : ae_fin_strongly_measurable f μ) :
+  ae_measurable f μ :=
+⟨hf.mk f, hf.fin_strongly_measurable_mk.measurable, hf.ae_eq_mk⟩
+
+protected lemma ae_measurable_ennreal {f : α → ℝ≥0∞} (hf : ae_fin_strongly_measurable f μ) :
+  ae_measurable f μ :=
+⟨hf.mk f, hf.fin_strongly_measurable_mk.measurable_ennreal, hf.ae_eq_mk⟩
+
+end mk
+
+section arithmetic
+
+protected lemma mul [monoid_with_zero β] [no_zero_divisors β] [has_continuous_mul β]
+  (hf : ae_fin_strongly_measurable f μ) (hg : ae_fin_strongly_measurable g μ) :
+  ae_fin_strongly_measurable (f * g) μ :=
+⟨hf.mk f * hg.mk g, hf.fin_strongly_measurable_mk.mul hg.fin_strongly_measurable_mk,
+  hf.ae_eq_mk.mul hg.ae_eq_mk⟩
+
 protected lemma add [add_monoid β] [has_continuous_add β]
   (hf : ae_fin_strongly_measurable f μ) (hg : ae_fin_strongly_measurable g μ) :
   ae_fin_strongly_measurable (f + g) μ :=
-⟨hf.some + hg.some, hf.some_spec.1.add hg.some_spec.1, hf.some_spec.2.add hg.some_spec.2⟩
+⟨hf.mk f + hg.mk g, hf.fin_strongly_measurable_mk.add hg.fin_strongly_measurable_mk,
+  hf.ae_eq_mk.add hg.ae_eq_mk⟩
 
 protected lemma neg [add_group β] [topological_add_group β] (hf : ae_fin_strongly_measurable f μ) :
   ae_fin_strongly_measurable (-f) μ :=
-⟨-hf.some, hf.some_spec.1.neg, hf.some_spec.2.neg⟩
+⟨-hf.mk f, hf.fin_strongly_measurable_mk.neg, hf.ae_eq_mk.neg⟩
 
 protected lemma sub [add_group β] [has_continuous_sub β]
   (hf : ae_fin_strongly_measurable f μ) (hg : ae_fin_strongly_measurable g μ) :
   ae_fin_strongly_measurable (f - g) μ :=
-⟨hf.some - hg.some, hf.some_spec.1.sub hg.some_spec.1, hf.some_spec.2.sub hg.some_spec.2⟩
+⟨hf.mk f - hg.mk g, hf.fin_strongly_measurable_mk.sub hg.fin_strongly_measurable_mk,
+  hf.ae_eq_mk.sub hg.ae_eq_mk⟩
+
+protected lemma const_smul {𝕜} [semiring 𝕜] [topological_space 𝕜] [add_comm_monoid β] [module 𝕜 β]
+  [no_zero_smul_divisors 𝕜 β] [has_continuous_smul 𝕜 β]
```

Comment:
Is the assumption `no_zero_smul_divisors 𝕜 β` relevant here? I expect that having zero divisors will just make the function even more zero, so the finite support condition should still be satisfied.

Reply:
Indeed it's not useful. Fixed.